sorald/src/main/java/sorald/processor/MathOnFloatProcessor.java

```diff
@@ -56,12 +56,8 @@ protected void repairInternal(CtBinaryOperator element) {
                      */
                     binaryOperator.setType(getFactory().Type().doublePrimitiveType());
                 }
-                if (!binaryOperator.getType().equals(getFactory().Type().doublePrimitiveType())) {
-                    binaryOperator
-                            .getRightHandOperand()
-                            .setTypeCasts(List.of(getFactory().Type().doublePrimitiveType()));
-                    binaryOperator.setType(getFactory().Type().doublePrimitiveType());
-                }
+                // We do not need to cast the type of the right hand operand as it is already a
+                // double
```

Comment:
@monperrus not related to autocompletion, but when I deleted the above code, the inline comment wa suggested by co-pilot. 🤌🏼

Reply:
Unbelievable!

Reply:
crazy